RecoParticleFlow/PFClusterProducer/plugins/Basic2DGenericPFlowClusterizer.cc

```diff
@@ -166,10 +183,22 @@ growPFClusters(const reco::PFCluster& topo,
 	std::abs(refhit->positionREP().eta()) > 0.34 ) {
       cell_layer *= 100;
     }  
-    const double recHitEnergyNorm = 
-      _recHitEnergyNorms.find(cell_layer)->second; 
+
     math::XYZPoint topocellpos_xyz(refhit->position());
     dist2.clear(); frac.clear(); fractot = 0;
+
+    double recHitEnergyNorm=0.;
+    std::tuple<std::vector<int> ,std::vector<int> , std::vector<double> > _recHitEnergyNormsT = _recHitEnergyNorms.find(cell_layer)->second;
```

Comment:
all of these vector copies can be pretty costly.
Please provide a time per event or better yet a profiler analysis to assure that there is no CPU cost increase.

Reply:
Perhaps what the author is trying to do is make a Structure of Arrays? If so, we have two CMS specific classes for that
https://github.com/cms-sw/cmssw/blob/master/FWCore/SOA/interface/Table.h
and
https://github.com/cms-sw/cmssw/blob/master/FWCore/Utilities/interface/SoATuple.h

Reply:
ok, I agree that this whole vector manipulation can be improved.
I will try.

Reply:
for not please at least use a const ref here ```auto const& recHitEnergyNormsT = _recHitEnergyNorms.find(cell_layer)->second;```
 (and also drop the leading underscore in the local variable to reduce confusion with member data)

Reply:
ok also for this